buildscripts/checkstyle.xml

```diff
@@ -375,6 +375,14 @@
     <module name="SuppressWarningsHolder"/>
     <!-- this enables CHECKSTYLE:OFF and CHECKSTYLE:ON comments -->
     <module name="SuppressionCommentFilter"/>
+    <module name="Regexp">
+      <!-- see comment in otel.errorprone-conventions.gradle.kts for why we disable the SystemOut
+           rule there and rely on this check -->
+      <property name="id" value="SystemOut"/>
+      <!-- . matches any character, so we need to escape it and use \. to match dots. -->
+      <property name="format" value="System\.out\.println"/>
```

Comment:
I believe errorprone also detects `System.err.println()` and `printStackTrace()` with the same rule. It might be better to check for `print` instead of `println`.

Reply:
+1 for `print` to catch `printf` also

Reply:
Let's add Thread.dumpStack too